luda-editor/luda-editor-client/src/editor/mod.rs

```diff
@@ -112,6 +119,84 @@ impl namui::Entity for Editor {
+                        async move {
+                            fn handle_error(path: String, started_at: Duration, error: String) {
+                                namui::log(format!("error on read_file: {:?}", error));
+                                namui::event::send(Box::new(
+                                    EditorEvent::SequenceLoadStateUpdateEvent {
+                                        path,
+                                        state: Some(SequenceLoadState {
+                                            started_at,
+                                            detail: SequenceLoadStateDetail::Failed { error },
+                                        }),
+                                    },
+                                ));
+                            }
+                            let result = socket
+                                .read_file(luda_editor_rpc::read_file::Request {
+                                    dest_path: path.clone(),
+                                })
+                                .await;
+                            match result {
+                                Ok(response) => {
+                                    let file = response.file;
+                                    match Sequence::try_from(file) {
+                                        Ok(sequence) => namui::event::send(Box::new(
+                                            EditorEvent::SequenceLoadStateUpdateEvent {
+                                                path: path.clone(),
+                                                state: Some(SequenceLoadState {
+                                                    started_at,
+                                                    detail: SequenceLoadStateDetail::Loaded {
+                                                        sequence,
+                                                    },
+                                                }),
+                                            },
+                                        )),
+                                        Err(error) => handle_error(path.clone(), started_at, error),
+                                    }
+                                }
+                                Err(error) => handle_error(path.clone(), started_at, error),
+                            }
+                        }
+                    });
+                }
+                EditorEvent::SequenceLoadStateUpdateEvent { path, state } => match state {
+                    Some(state) => {
+                        if let Some(old_state) = self.sequence_load_state_map.get(path) {
+                            if old_state.started_at > state.started_at {
+                                return;
+                            }
+                        }
+
+                        self.sequence_load_state_map
+                            .insert(path.clone(), (*state).clone());
+                    }
+                    None => {
```

Comment:
What's the meaning of None for `Option<SequenceLoadState>`?

Reply:
It means `clearing LoadState`.
`clearing LoadState` will probably be necessary to make `sequence list view`.
In this map, key is path of sequence file. So if the path changes with action like renaming. Old path's state should be cleared.